### DISCUSSION TITLE
Help Needed: Connection Issues with Alpyne and AnyLogic PLE v8.8.5

### DISCUSSION THREAD

**Discussion by HerbertRolf** [the-anylogic-company/Alpyne#48] [Q&A] [answered]
Hello everyone,

I'm currently facing a challenge while trying to integrate Alpyne with an AnyLogic Model (PLE version 8.8.5) and was hoping someone here might have encountered a similar issue and found a solution.

**Issue Description**
Whenever I attempt to start a client using AlpyneClient, I encounter the following error message:

_ConnectionRefusedError: [WinError 10061] No connection could be made because the target computer actively refused  it_

This is followed by a cascade of exceptions culminating in a ModelError indicating a failure to start the app properly. The exact message is:

_ModelError: ModelError[path=, status=Failed to properly start the app. Error from process: , error=None, msg='']_

**Troubleshooting Done**

- When I use the updated Example Model Stock Management Game from Alpyne's Beta Release 0.2.0 ([link to release](https://github.com/t-wolfeadam/Alpyne/releases/tag/v0.2.0)), everything works as expected.

- The issue arises when I open this example model in AnyLogic, save it (even without making any changes), and then try to run it again through Alpyne. At this point, the connection error occurs.

**Questions**
1. Has anyone experienced a similar issue with Alpyne and AnyLogic integration?
2. Are there specific settings or configurations within AnyLogic that need to be adjusted to ensure compatibility with Alpyne?
3. Could this be related to how AnyLogic saves the model files, potentially altering something that affects Alpyne's ability to connect? 
 
I would greatly appreciate any insights or suggestions you might have. Thank you in advance for your help!

**Comment by HerbertRolf** [comment]
So apparently, the new versions of AnyLogic  (8.8.5 / 8.8.6) are not supporting the Alpyne API. I now use a previous version (8.7.7) and it works perfectly fine. I hope this gets fixed in a future update.

**Comment by t-wolfeadam** [comment, marked as answer]
To you and @HerbertRolf - I'm working on overhauling the library which is not backwards compatible and requires the latest version of AL (8.8.6), but should be much more stable to run and easier to use. 

It's currently available on a side branch, which is installable via the following command:

> pip install "anylogic-alpyne @ git+https://github.com/t-wolfeadam/Alpyne.git@server-overhaul-api-revisal"

If you want to include the dependencies needed for the examples use this:

> pip install "anylogic-alpyne[examples] @ git+https://github.com/t-wolfeadam/Alpyne.git@server-overhaul-api-revisal"

And the examples themselves can be accessed by cloning the branch or [downloading the zip](https://github.com/t-wolfeadam/Alpyne/archive/refs/heads/server-overhaul-api-revisal.zip).

Also, the docs are updated but not uploaded as the pages, so I've built a singlehtml version for convenience, which you can download here:  [alpyne-v1rc-docs-singlehtml.zip](https://github.com/t-wolfeadam/Alpyne/files/14438408/alpyne-v1rc-docs-singlehtml.zip)

**Reply by HerbertRolf** [reply]
Thanks @t-wolfeadam  for the great work on Alpyne 1.0.0. The update brings a lot of improvements, but I've encountered an issue with type conversion that wasn't present in Alpyne 0.2.0.
When initializing the simulation with sim.reset(initialWorker=initialWorker) where initialWorker is a list of lists (e.g., initialWorker = [[1,1,0], [0,1,0], [1,0,1], [1,0,0], [1,0,1], [0,0,1]]), I receive an AssertionError: Unhandled type conversion. This setup worked seamlessly in version 0.2.0, suggesting a regression or change in how list of lists are handled.
The simulation expects an int[][] type for the corresponding variable, which matches Python's list[int[]] structure. This mismatch seems to trigger the error.
Could you provide any guidance on how to address this issue? Are there any workarounds or adjustments I can make in my code to accommodate the changes in version 1.0.0? Additionally, if there's any more information or specific error logs that would help diagnose this issue, I'm happy to provide them.
Thank you for your assistance and for the continued development of Alpyne.

**Reply by t-wolfeadam** [reply]
My pleasure! It's long overdue and I appreciate the patience. Generally, any problem you run into, providing the logs (e.g., alpyne.log / model.log), scripts (even snippets), and model (when possible) will help to expedite getting the solution. Though, I happened to have the time to try to recreate the problem and found a likely cause.

For helping others who might encounter a similar issue in the future, and for organization purposes, I've opened a new issue on the topic and will post my follow-up there: see #51
